plugins/Marketplace/PackageModel.py

```diff
@@ -250,3 +277,148 @@ def isCompatibleMaterialStation(self) -> bool:
     @pyqtProperty(bool, constant = True)
     def isCompatibleAirManager(self) -> bool:
         return self._is_compatible_air_manager
+
+    # --- manage buttons signals ---
+
+    stateManageButtonChanged = pyqtSignal()
+
+    installPackageTriggered = pyqtSignal(str)
+
+    uninstallPackageTriggered = pyqtSignal(str)
+
+    updatePackageTriggered = pyqtSignal(str)
+
+    enablePackageTriggered = pyqtSignal(str)
+
+    disablePackageTriggered = pyqtSignal(str)
+
+    recentlyInstalledChanged = pyqtSignal(bool)
+
+    # --- enabling ---
+
+    @pyqtProperty(str, notify = stateManageButtonChanged)
+    def stateManageEnableButton(self) -> str:
+        """The state of the manage Enable Button of this package"""
+        if self._is_enabling == ManageState.PROCESSING:
+            return "busy"
+        if self._package_type == "material" or not self._is_installed:
+            return "hidden"
+        if self._is_installed and self._is_active:
+            return "secondary"
+        return "primary"
+
+    @property
+    def is_enabling(self) -> ManageState:
+        """Flag if the package is being enabled/disabled"""
+        return self._is_enabling
+
+    @is_enabling.setter
+    def is_enabling(self, value: ManageState) -> None:
+        if value != self._is_enabling:
+            self._is_enabling = value
+            if value == ManageState.HALTED:
+                self._is_recently_enabled = True
+            self.stateManageButtonChanged.emit()
+
+    @property
+    def is_active(self) -> bool:
+        """Flag if the package is currently active"""
+        return self._is_active
+
+    @is_active.setter
+    def is_active(self, value: bool) -> None:
+        if value != self._is_active:
+            self._is_active = value
+            self.stateManageButtonChanged.emit()
+
+    # --- Installing ---
+
+    @pyqtProperty(str, notify = stateManageButtonChanged)
+    def stateManageInstallButton(self) -> str:
+        """The state of the Manage Install package card"""
+        if self._is_installing == ManageState.PROCESSING:
+            return "busy"
+        if self._is_recently_installed:
+            return "confirmed"
+        if self._is_installed:
+            if self._is_bundled and not self._can_downgrade:
+                return "hidden"
+            else:
+                return "secondary"
+        else:
+            return "primary"
+
+    @property
+    def is_installing(self) -> ManageState:
+        """Flag is we're currently installing, when setting this to ``None`` in indicates a failed installation"""
+        return self._is_installing
+
+    @is_installing.setter
+    def is_installing(self, value: ManageState) -> None:
+        if value != self._is_installing:
+            self._is_installing = value
+            if value == ManageState.HALTED:
+                self._is_recently_installed = True
+            self.stateManageButtonChanged.emit()
+
+    @property
+    def is_recently_installed(self):
```

Comment:
What does recently mean? During this boot of Cura?

Reply:
Yes, should I work that into the naming?